batch/batch/batch.py

```diff
@@ -1118,36 +1110,31 @@ def __anext__(self):
 async def refresh_k8s_pods():
     # if we do this after we get pods, we will pick up jobs created
     # while listing pods and unnecessarily restart them
-    pod_jobs = [Job.from_record(record) for record in await db.jobs.get_records_where({'pod_name': 'NOT NULL'})]
+    pod_jobs = [Job.from_record(record) for record in await db.jobs.get_records_where({'state': ('Ready', 'Running')})]
 
     pods, err = await app['k8s'].list_pods(
         label_selector=f'app=batch-job,hail.is/batch-instance={INSTANCE_ID}')
     if err is not None:
         traceback.print_tb(err.__traceback__)
-        log.info('could not refresh pods due to {err}, will try again later')
+        log.info(f'could not refresh pods due to {err}, will try again later')
         return
 
     log.info(f'k8s had {len(pods.items)} pods')
 
     seen_pods = set()
     for pod in pods.items:
-        if pod.metadata.name is None:
-            log.info(f'ignoring pod.metadata.name is None: {pod.to_dict()}')
-            continue
-
         pod_name = pod.metadata.name
         seen_pods.add(pod_name)
 
-        job = Job.from_record(await db.jobs.get_record_by_pod(pod_name))
+        job = await Job.from_k8s_labels(pod)
         if job and not job.is_complete():
             await update_job_with_pod(job, pod)
 
-    log.info('starting pods not seen in k8s')
+    log.info('restarting ready and running jobs with pods not seen in k8s')
 
     for job in pod_jobs:
-        pod_name = job._pod_name
-        if pod_name not in seen_pods:
-            log.info(f'restarting job {job.id}')
+        if job._pod_name not in seen_pods:
+            log.info(f'restarting job {job.id}, task {job._current_task.name!r}')
```

Comment:
you do this `job {job.id}, task {job._current_task.name!r}` in a bunch of places, probably worth giving that a name like `job_task_full_name` or something.